File: backend/src/main/java/com/smartcampus/backend/dto/weather/WeatherAtDTO.java
package com.smartcampus.backend.dto.weather;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class WeatherAtDTO {

    private Double latitude;
    private Double longitude;
    private LocalDate date;
    private LocalTime time;

    // Resume du jour demande
    private Double temperatureMax;
    private Double temperatureMin;
    private Double precipitationSum;
    private Double windspeedMax;

    // Point horaire demande (rempli seulement si "time" est fourni et disponible)
    private Double temperature;
    private Integer precipitationProbability;

    // Code meteo + traduction (icone / signification)
    private Integer weathercode;
    private String description;
    private String icon;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/weather/WeatherAtDto.kt
package com.ObservatoireCampus.mobile.model.weather

data class WeatherAtDto(
    val latitude: Double?,
    val longitude: Double?,
    val date: String?,
    val time: String?,
    val temperatureMax: Double?,
    val temperatureMin: Double?,
    val precipitationSum: Double?,
    val windspeedMax: Double?,
    val temperature: Double?,
    val precipitationProbability: Int?,
    val weathercode: Int?,
    val description: String?,
    val icon: String?
)